crates/loam-cli/src/commands/build/build_clients.rs

```diff
@@ -123,79 +125,57 @@ impl Args {
 
         Ok(())
     }
-     
-    fn get_contract_alias(&self, name: &str) -> Result<Option<String>, Error> {
-        let config_dir = cli::config::locator::Args {
-            global: false,
-            config_dir: None,
-        }.config_dir()?;
-        let alias_file = config_dir.join("contract-ids").join(format!("{}.json", name));
-        
-        if alias_file.exists() {
-            let content = std::fs::read_to_string(alias_file)?;
-            let alias_data: serde_json::Value = serde_json::from_str(&content)?;
-            let network_passphrase = std::env::var("STELLAR_NETWORK_PASSPHRASE")
-                .expect("No STELLAR_NETWORK_PASSPHRASE environment variable set");
-            
-            if let Some(id) = alias_data["ids"].get(&network_passphrase) {
-                Ok(Some(id.as_str().unwrap().to_string()))
-            } else {
-                Ok(None)
-            }
-        } else {
-            Ok(None)
-        }
-    }
 
-    async fn contract_hash_matches(&self, contract_id: &str, hash: &str, network: &Network) -> Result<bool, Error> {
-        let network_args = cli::network::Args {
+
+    fn get_network_args(network: &Network) -> cli::network::Args {
```

Comment:
I think we should instead remove the local `Network` type, which should have never existed since this type exists.

Reply:
Okay, should I perform the refactor now or in a separate PR?

Reply:
I would go ahead and do it now since you need the original type.

Reply:
I have the refactor done but the deserialize is not implemented for cli::network::Args which is needed in `Environment`:

```
#[derive(Debug, serde::Deserialize, Clone)]
pub struct Environment {
    pub accounts: Option<Vec<Account>>,
    pub network: Network,
    pub contracts: Option<Map<Box<str>, Contract>>,
}
```
It looks like this could be worked around with a custom deserialize method for Environment, if you want to go that route.

Reply:
There are actually two Network types in the CLI. Here is the the one that is serialized: https://github.com/stellar/stellar-cli/blob/7984cd0d014d67ff3ca5fe720b615001b255d524/cmd/soroban-cli/src/commands/network/mod.rs#L170

The other is just for the CLI args. The idea is that that is used to resolve to a network with a `rpc_url` and `network_passphrase`, no options.

Reply:
Excellent, thanks

Reply:
Problem I'm seeing is that `Network` doesnt have the `network` argument like the `Args` version does. The `name` parameter in environments.toml is being mapped to `network`.

Reply:
Stop @willemneal! I already explained why we need our own `Network`! https://github.com/loambuild/loam-sdk/pull/42#pullrequestreview-2130862138

Reply:
I seem to have rediscovered this reasoning lol